drivers/bme680/bme680.c

```
@@ -0,0 +1,153 @@
+    if (params->ifsel == BME680_I2C_INTF) {
+#ifdef MODULE_BME680_I2C
+        BME680_SENSOR(dev).intf = BME680_I2C_INTF;
+        BME680_SENSOR(dev).read = bme680_i2c_read_hal;
+        BME680_SENSOR(dev).write = bme680_i2c_write_hal;
+#else
+        LOG_ERROR("[bme680]: module bme680_i2c not enabled\n");
+        return BME680_NO_DEV;
+#endif
+    }
+    else {
+#ifdef MODULE_BME680_SPI
+        BME680_SENSOR(dev).intf = BME680_SPI_INTF;
+        BME680_SENSOR(dev).read = bme680_spi_read_hal;
+        BME680_SENSOR(dev).write = bme680_spi_write_hal;
+        spi_init_cs(SPI_DEV(0), params->intf.spi.nss_pin);
+#else
+        LOG_ERROR("[bme680]: module bme680_spi not enabled\n");
+        return BME680_NO_DEV;
+#endif
+    }
+
+    BME680_SENSOR(dev).delay_ms = bme680_ms_sleep;
+
+    /* call internal bme680_init from Bosch Sensortech driver */
+    ret = bme680_init_internal(&BME680_SENSOR(dev));
+    if (ret != 0) {
+        DEBUG("[bme680]: Failed to get ID\n");
+        return -1;
+    }
+
+    /*  retrieve params and set them in bme680_t */
+    BME680_SENSOR(dev).tph_sett.os_temp = params->temp_os;
+    BME680_SENSOR(dev).tph_sett.os_hum = params->hum_os;
+    BME680_SENSOR(dev).tph_sett.os_pres = params->pres_os;
+
+    BME680_SENSOR(dev).tph_sett.filter = params->filter;
+
+    /* Enable gas measurement if needed */
+    BME680_SENSOR(dev).gas_sett.run_gas = params->gas_measure;
+    /* Create a ramp heat waveform in 3 steps */
+    BME680_SENSOR(dev).gas_sett.heatr_temp = params->heater_temp;
+    BME680_SENSOR(dev).gas_sett.heatr_dur = params->heater_dur;
+
+    /* Select the intended power mode */
+    /* Must be set before writing the sensor configuration */
+    BME680_SENSOR(dev).power_mode = BME680_FORCED_MODE;
+
+    /* Set the desired sensor configuration */
+    ret = bme680_set_sensor_settings(params->settings, &BME680_SENSOR(dev));
+    if (ret != 0) {
+        DEBUG("[bme680]: failed to set settings\n");
+        return -2;
```

```suggestion
        return BME680_COM_FAILED;
```
?

---

We can simply return the error code as returned  from the vendor driver.